framework/src/modules/chain/chain.js

```diff
@@ -188,6 +188,89 @@ module.exports = class Chain {
 		}
 	}
 
+	get actions() {
+		return {
+			calculateSupply: action => blockReward.calcSupply(action.params[0]),
+			calculateMilestone: action => blockReward.calcMilestone(action.params[0]),
+			calculateReward: action => blockReward.calcReward(action.params[0]),
+			generateDelegateList: action =>
+				new Promise((resolve, reject) => {
+					this.scope.modules.delegates.generateDelegateList(
+						action.params[0],
+						action.params[1],
+						(err, data) => {
+							if (err) {
+								reject(err);
+							}
+
+							resolve(data);
+						},
+						action.params[2]
+					);
+				}),
+			getNetworkHeight: async action =>
+				util.promisify(this.scope.modules.peers.networkHeight)(
+					action.params[0]
+				),
+			getAllTransactionsCount: async () =>
+				util.promisify(
```

Comment:
promisify in the top of the file?

Reply:
Will tackle this in https://github.com/LiskHQ/lisk/issues/2991